watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -222,28 +387,46 @@ def eq_recovery_mass(b, t):
             doc="Average thermal efficiency",
         )
         def thermal_efficiency(b, t):
-            total_enth_flux = sum(
-                b.flux_enth_hot[t, x] for x in self.difference_elements
-            )
-            total_cond_heat_flux = sum(
-                b.flux_conduction_heat[t, x] for x in self.difference_elements
-            )
-
-            return total_enth_flux / (total_enth_flux + total_cond_heat_flux)
-
-        @self.Expression(
-            self.flowsheet().config.time,
-            doc="module heat recovery: ratio of the actual heat recovered by cold side to the maximum ideal heat recovery",
-        )
-        def effectiveness(b, t):
+            if self.config.MD_configuration_Type in [
+                MDconfigurationType.DCMD,
+                MDconfigurationType.PGMD_CGMD,
+                MDconfigurationType.AGMD,
+            ]:
+
+                total_enth_flux = sum(
+                    b.flux_enth_hot[t, x] for x in self.difference_elements
+                )
+                total_cond_heat_flux = sum(
+                    b.flux_conduction_heat[t, x] for x in self.difference_elements
+                )
 
-            return (
-                b.cold_ch.properties[t, b.first_element].temperature
-                - b.cold_ch.properties[t, b.cold_ch.length_domain.last()].temperature
-            ) / (
-                b.hot_ch.properties[t, b.first_element].temperature
-                - b.cold_ch.properties[t, b.cold_ch.length_domain.last()].temperature
+                return total_enth_flux / (total_enth_flux + total_cond_heat_flux)
+            else:
+                return Constraint.Skip
```

Comment:
Why do we skip for VMD?

Reply:
Heat recovery (effectiveness) is not feasible for a single VMD module unless it is part of a multi-stage or multi-effect configuration. Thermal efficiency is highest for VMD